src/main/java/com/rtb/projectmanagementtool/user/UserController.java

```diff
@@ -2,19 +2,27 @@
 
 import com.google.appengine.api.datastore.*;
 import com.google.appengine.api.datastore.Query.*;
-import java.util.HashSet;
+import java.util.ArrayList;
 
 public final class UserController {
-  private HashSet<UserData> users;
 
-  public UserController(HashSet<UserData> users) {
-    this.users = users;
+  private DatastoreService datastore;
+
+  public UserController(DatastoreService datastore) {
+    this.datastore = datastore;
+  }
+
+  public UserData getUserByID(long userID) {
+    Query query = new Query("User").addFilter("userID", FilterOperator.EQUAL, userID);
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    UserData user = new UserData(entity);
```

Comment:
The null pointer exception, this piece here is probably returning a null entity.  My guess is the query by ID didn't return a recrod, and asSingleEntity returned null.